src/main/java/com/limechain/trie/cache/node/PendingInsertUpdate.java

```diff
@@ -0,0 +1,40 @@
+package com.limechain.trie.cache.node;
+
+import com.limechain.runtime.version.StateVersion;
+import com.limechain.trie.structure.nibble.Nibbles;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public record PendingInsertUpdate(byte[] newMerkleValue,
+                                  List<byte[]> childrenMerkleValues,
+                                  Nibbles partialKey,
+                                  StateVersion stateVersion,
+                                  @Nullable byte[] value) implements PendingTrieNodeChange {
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        PendingInsertUpdate that = (PendingInsertUpdate) o;
+        return Objects.equals(partialKey, that.partialKey)
+            && Objects.deepEquals(newMerkleValue, that.newMerkleValue)
+            && Objects.equals(childrenMerkleValues, that.childrenMerkleValues);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(Arrays.hashCode(newMerkleValue), partialKey, childrenMerkleValues);
+    }
+
+    @Override
+    public String toString() {
+        return "PendingInsertUpdate{" +
+            "newMerkleValue=" + Arrays.toString(newMerkleValue) +
+            ", partialKey=" + partialKey +
+            ", childrenMerkleValues=" + childrenMerkleValues +
+            '}';
+    }
```

Comment:
None of these can be switched with a lombok annotation? `@EqualsAndHasCode`, `@ToString`

Reply:
Lombok annotation are not usable on java records.